src/3d/qgs3dmapsettings.h

```diff
@@ -74,6 +74,22 @@ class _3D_EXPORT Qgs3DMapSettings : public QObject, public QgsTemporalRangeObjec
     //! Resolves references to other objects (map layers) after the call to readXml()
     void resolveReferences( const QgsProject &project );
 
+    /**
+     * Returns the 3D scene's 2D extent in project's CRS
+     * \since QGIS 3.30
+     */
+    QgsRectangle extent() const { return mExtent; }
+
+    /**
+     * Sets the 3D scene's 2D \a extent in project's CRS, while also setting the scene's origin to the extent's center
+     * This needs to be called during initialization, as terrain will only be generated
+     * within this extent and layer 3D data will only be loaded within this extent too.
```

Comment:
Is this still the case?

Reply:
This is actually one of the introduced changes. We now need to explicitly set the scene's extent in `Qgs3dMapSettings`, then the terrain generators and 3d layers will honor this extent and only load data that falls within it.